### PR TITLE
bump to scarb 2.12.2

### LINKED ISSUE #345: Upgrade Cairo to 2.10.1
https://github.com/starkware-libs/cairo/releases/tag/v2.10.1

### PR DESCRIPTION
**Pull request by 0xNeshi**
Closes #345 

Related https://github.com/exercism/cairo-test-runner/pull/36

### CODE REVIEW

**Comment by 0xNeshi** [comment]
Found [breaking issues](https://github.com/exercism/cairo/actions/runs/16730492719/job/47357109676?pr=369) with the 2.12.0 version when updating the Cairo track, putting the PR into draft until fix is introduced

**Comment by 0xNeshi** [comment]
Still blocked: https://github.com/starkware-libs/cairo/issues/8467

**Comment by 0xNeshi** [comment]
For anyone reviewing this PR, the changes in this PR are major enough that they should [cause a rerun of all students' tests](https://github.com/exercism/cairo/pull/369#issuecomment-3151777114).

File: exercises/practice/simple-linked-list/.meta/example.cairo [hunk 1]
```diff
@@ -6,12 +6,14 @@ pub struct SimpleLinkedList<T> {
...
 type List<T> = Option<Box<Node<T>>>;
 
-#[derive(Drop, Copy)]
 struct Node<T> {
     data: T,
     next: List<T>,
 }
 
+impl NodeDrop<T, +Drop<T>> of Drop<Node<T>>;
+impl NodeCopy<T, +Copy<T>> of Copy<Node<T>>;
```
Comment (0xNeshi) [inline exercises/practice/simple-linked-list/.meta/example.cairo]: This is a necessary change introduced with the bumped version

**Comment by 0xNeshi** [comment]
> This PR touches files which potentially affect the outcome of the tests of an exercise. This will cause all students' solutions to affected exercises to be re-tested.
> 
> If this PR does **not** affect the result of the test (or, for example, adds an edge case that is not worth rerunning all tests for), **please add the following to the merge-commit message** which will stops student's tests from re-running. Please copy-paste to avoid typos.
> 
> ```
> [no important files changed]
> ```
> 
> For more information, refer to the [documentation](https://exercism.org/docs/building/tracks#h-avoiding-triggering-unnecessary-test-runs). If you are unsure whether to add the message or not, please ping `@exercism/maintainers-admin` in a comment. Thank you!

It is necessary to re-run students' exercises, as their solutions might not work with the new Cairo version (e.g. see implementation and formatting changes in [simple-linked-list](https://github.com/exercism/cairo/pull/369/files#diff-7b1b575b978cf5d389ec044caed69cc107ca0aaf167ca7ce35bf77d38431e270)).
